pkg/lifecycle/state.go

```diff
@@ -0,0 +1,136 @@
+package lifecycle
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/go-git/go-billy/v5"
+	"github.com/rancher/charts-build-scripts/pkg/filesystem"
+	"github.com/rancher/charts-build-scripts/pkg/path"
+	"github.com/sirupsen/logrus"
+)
+
+// SaveState will save the lifecycle-status state to state.json file at charts repo
+func (s *Status) SaveState() error {
+	logrus.Info("saving app state to state.json file")
+
+	// Marshal the Project struct into JSON
+	data, err := json.MarshalIndent(s, "", "  ")
+	if err != nil {
+		err = fmt.Errorf("error marshalling project to JSON: %w", err)
```

Comment:
Are all of the errors here lacking enough context to warrant having more info added? Usually not necessary and not a standard for the lang or the repo.